src/main/java/de/tum/in/www1/artemis/service/connectors/ci/AbstractContinuousIntegrationResultService.java

```diff
@@ -102,7 +106,8 @@ private void addStaticCodeAnalysisFeedbackToResult(Result result, AbstractBuildR
     }
 
     private void addTestwiseCoverageReportToResult(Result result, AbstractBuildResultNotificationDTO buildResult, ProgrammingExercise programmingExercise) {
-        if (Boolean.TRUE.equals(programmingExercise.isTestwiseCoverageEnabled())) {
+        programmingExercise = programmingExerciseRepository.getProgrammingExerciseWithBuildConfigElseThrow(programmingExercise);
+        if (Boolean.TRUE.equals(programmingExercise.getBuildConfig().isTestwiseCoverageEnabled())) {
```

Comment:
Why do you reload the exercise again if you only need the configuration? Would it make sense to directly load the programming exercise with its configuration wherever the value for the parameter is loaded in our code?

Reply:
This method is used in many different places. In some cases, the original value is retrieved through the participation, where we would have to load it manually anyway. And the value is not needed in the majority of the process